Change number format default

khtruong opened the pull request:
We have gotten feedback that the default number format (.3s) is confusing, especially if the values are less than 1 (i.e. 0.125 = 125m). The ask was to change the default number format. Let me know if this is undesirable.

john-bodley committed:
[csv-upload] Fixing message encoding (#6971)


(cherry picked from commit 48431ab5b9375a94c5262a0336d9c69e5f01a3ac)

conglei committed:
Enhancement of query context and object. (#6962)

* added more functionalities for query context and object.

* fixed cache logic

* added default value for groupby

* updated comments and removed print

(cherry picked from commit d5b9795f87f79fa2c41e144ffc00fd9586be7657)

khtruong committed:
Sparkline dates aren't formatting in Time Series Table (#6976)

* Exclude venv for python linter to ignore

* Fix NaN error

john-bodley committed:
[sql-parse] Fixing LIMIT exceptions (#6963)


(cherry picked from commit 3e076cb60b385e675ed1c9a8053493375e43370b)

michellethomas committed:
Adding custom control overrides (#6956)

* Adding extraOverrides to line chart

* Updating extraOverrides to fit with more cases

* Moving extraOverrides to index.js

* Removing webpack-merge in package.json

* Fixing metrics control clearing metric

(cherry picked from commit e6194051f486e42922dc4e34a861f4490c1062fc)

Commit:
[fix] /superset/slice/id url is too long (#6989)


(cherry picked from commit 6a4d507ab607b01ed324cb3341b71c6fb2cb5c97)

michellethomas committed:
Changing time table viz to pass formatTime a date (#7020)

(cherry picked from commit 7f3c145b1f5a4e2d8b95982119503e98772e2c47)

xtinec committed:
Fix the white background shown in SQL editor on drag (#7021)

This PR sets the background-color css property on `.ace_scroller` instead of `.ace_content` to prevent the white background shown during resizing of the SQL editor before drag ends.

john-bodley committed:
[db-engine-spec] Aligning Hive/Presto partition logic (#7007)


(cherry picked from commit 05be86611785fef2904992e4e7d31dce23f1c51b)

betodealmeida committed:
Show tooltip with time frame (#6979)

betodealmeida committed:
Fix time filter control (#6978)

thunter009 committed:
[WIP] fix user specified JSON metadata not updating dashboard on refresh (#7027)


(cherry picked from commit cc58f0e661044e95c7c86d0da8d77a0a6640efe7)

conglei committed:
fix inaccurate data calculation with adata rolling and contribution (#7035)


(cherry picked from commit 0782e831cd37f665a2838119d87c433269f1b36b)

john-bodley committed:
[sqlparse] Fixing table name extraction for ill-defined query (#7029)


(cherry picked from commit 07c340cf8203f13222f16efad1e55e202deb1865)

Commit:
[fix] Cursor jumping when editing chart and dashboard titles (#7038)


(cherry picked from commit fc1770f7b79a4d8815b646b46390fabf190c3815)

Commit:
[fix] explore chart from dashboard missed slice title (#7046)


(cherry picked from commit a6d48d4052839286aec725d51303b3b2bf6e8dd4)

john-bodley committed:
[datasource] Ensuring consistent behavior of datasource editing/saving. (#7037)

* Update datasource.py

* Update datasource.py

(cherry picked from commit c771625f1068d3a7f41e6bced14b0cbdbf9962cc)

michellethomas committed:
Adding warning message for sqllab save query (#7028)


(cherry picked from commit ead3d48133e7e1ab8b91d51e561544a544b4eaad)

Commit:
handle null column_name in sqla and druid models

(cherry picked from commit 2ff721ae072b8d69c5cabddc3e1a388a596b1b6f)

khtruong committed:
feat: add ability to change font size in big number (#7003)

* Add ability to change font sizes in Big Number

* rename big number to header

* Add comment to clarify font size values

betodealmeida committed:
Allow LIMIT to be specified in parameters (#7052)

john-bodley committed:
[missing values] Removing replacing missing values (#4905)


(cherry picked from commit 61add606ca16a6ba981ccde864b121f5464b697a)

Commit:
[forms] Fix handling of NULLs

(cherry picked from commit e83a07d3dfda350cc44041cb6cbaec4510887902)

mistercrunch committed:
Fix filter_box migration PR #6523 (#7066)

* Fix filter_box migration PR #6523

* Fix druid-related bug

(cherry picked from commit b210742ad24d01ca05bc58ca3342c90e301fe073)

enricoberti committed:
[SQL Lab] Improved query and results tabs rendering reliability (#7082)

closes #7080

(cherry picked from commit 9b58e9f4920ef424e5b545dcbb4726e22bed5982)

michellethomas committed:
Use metric name instead of metric in filter box (#7106)


(cherry picked from commit 003364e74ea70cad1a4a6e784933fe8bef4c78ec)

xtinec committed:
SQL editor layout makeover (#7102)

This PR includes the following layout and css tweaks:
- Using flex to layout the north and south sub panes of query pane so resizing works properly in both Chrome and Firefox
- Removal of necessary wrapper divs and tweaking of css in sql lab so we can scroll to the bottom of both the table list and the results pane
- Make sql lab's content not overflow vertically and layout the query result area to eliminate double scroll bars
- css tweaks on the basic.html page so the loading animation appears in the center of the page across the board

(cherry picked from commit 71f1bbd2ec59b99d6ba6d9a4a2f9cfceaf922b80)

mistercrunch committed:
Bump python lib croniter to an existing version (#7132)

Package maintainers should really never delete packages, but it appears
this happened with croniter and resulted in breaking our builds.

This PR bumps to a more recent existing version of the library

(cherry picked from commit 215ed392a11598eac228f57341dbfd232cf770e3)

khtruong committed:
Change number format default

mistercrunch commented:
I think @kristw did some work o altering the default one, I think the `G`  instead of `B` for billions was similarly confusing.

kristw commented:
@khtruong Change it to `SMART_NUMBER` will address both yours and @mistercrunch 's need.

kristw commented:
related to #6916

khtruong commented:
Thanks guys. Hopefully soon I'll get to the point where I will know about these already-done enhancements! 🙏

I'll change this PR so it goes into master :)

khtruong committed:
Use smart formatter instead

DiggidyDave committed:
Injectable statsd client (#7138)

* Add ability to inject statsd client; some py test/reqs updates

- Updated the metrics logger to allow construction with an existing
statsd client, so that it can be configured by external systems or libs.
- added requirements to requirements-dev.txt which are needed to run
  tests-eg coverage, nose
- removed dependency on mock lib, it is in python stdlib now
- updated tox.ini to remove the now-superfluous deps

* add license to test file, and remove blank line at EOF

(cherry picked from commit ba19a62412aba27d9c38ec7013f13a976cb6c054)

mistercrunch committed:
[bugfix] SQL Lab 'Filter Results' doesn't stick (#7104)

When using a "Search Results" criteria, the subset of rows that match
the criteria get displayed. While this the filter is applied, if another
query is run, the filter is still active, but not displayed in the input
text box. After this change, the state of the input box sticks after
subsequent queries.

(cherry picked from commit d5e8d663dc63ee9b99b3e7bb4e91512fa32ed1e0)

betodealmeida committed:
Revert PR #6933 (#7162)

khtruong committed:
[Lyft-GA] Enable color consistency in a dashboard (#7135)

* Enable color consistency in a dashboard

Moved actions, minor UI, allowed dashboard copy

Fix linting errors

Undo unintentional change

Updated and added unit tests

Fail quietly if package has not been updated

Fail quietly on dashboard copy if package is old

* Update packages

* Remove unnecessary code

* Addressed Grace's comments

* Small fix for item key

* Reset chart's color during exploration

* Do not reset chart form data when exploring chart

xtinec committed:
Fix double scroll bars when content of sql result table overflows horizontally (#7168)

The PR substracts the scrollbar height from the height of the container of the react virtualized table so we don't see double scrollbars.

(cherry picked from commit 7ffcabd5c08bbe6aa022e66590c0ce805ae13f08)

khtruong commented:
Unfortunately, the team decided to push all changes into lyftga to alleviate any pain points that could occur from cherry picking changes from master into lyftga and then merging lyftga back into master. The good news is that we should be merging lyftga into master in a week and a half. So this will be in master soon!

khtruong committed:
Merge branch 'lyftga' into VIZ-213

khtruong committed:
fix merge issues

kristw commented on superset/assets/backendSync.json:
```diff
@@ -1635,7 +1635,7 @@
       "label": "Number format",
       "renderTrigger": true,
-      "default": ".3s",
+      "default": "smart_number",
```
If you use `SMART_NUMBER` (all caps), this will be already registered out of the box and you don't have to do anything in `superset/assets/src/setup/setupFormatters.js`

khtruong replied:
My bad, I misunderstood. It's fixed now!

khtruong committed:
Use SMART_NUMBER
